openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -121,13 +122,20 @@ public void execute(ToolContext toolContext) {
             // Handle the code generation
             String oasFilePath = toolContext.filePath();
             Path packagePath = toolContext.currentPackage().project().sourceRoot();
-            Optional<OpenAPI> openAPI = getOpenAPIContract(packagePath, Path.of(oasFilePath), location, toolContext);
+            boolean isSanitized = false;
+            Map<String, ToolContext.Option> options = toolContext.options();
```

Comment:
We can separate this as a function

Reply:
move into the `getOpenAPIContract` function since we can take details using `toolContex`